src/pages/solvers/SAT_Solver.tsx

```diff
@@ -0,0 +1,38 @@
+import Head from "next/head";
+import React from "react";
+import type { NextPage } from "next";
+import { Container } from "../../components/Container";
+import { Main } from "../../components/Main";
+import { SolverTitle } from "../../components/solverComponents/SolverTitle";
+import { SAT_TextArea } from "../../components/solverComponents/SAT_Components/SAT_TextArea";
+import { SAT_Help } from "../../components/solverComponents/SAT_Components/SAT_Help";
+import { InputButtonPanel } from "../../components/solverComponents/InputButtonPanel";
+import { ProgressHandler } from "../../components/solverComponents/ProgressHandler";
+import { Divider } from "@chakra-ui/react";
+
+const SAT_Solver: NextPage = () => {
+  let SAT_TextAreaRef: React.MutableRefObject<SAT_TextArea | null> = React.createRef();
+  return (
+    <Container minHeight="100vh">
+      <Head>
+        <title>ProvideQ</title>
+        <meta name="description" content="Generated by create next app" />
+        <link rel="icon" href="/favicon.ico" />
+        {/* TODO: replace favicon */}
+      </Head>
+      <SolverTitle title="SAT Solver" text="For a given Boolean formula, this algorithm checks if there exists an interpretation that satisfies it." />
+      <Main mb="20vh">
+      <SAT_TextArea ref={SAT_TextAreaRef} />
```

Comment:
avoid using ref, never access the state of a child component directly in react! If state data is required on a higher level, you should declare the state in the higher component. Child components can access this state if the parent passes it as a prop to the child. Child components can change the state indirectly by firing events (accepting event handlers as props), if the parent subscribes to this event and updates the state accordingly.

The React manual has a [Lifting State Up](https://reactjs.org/docs/lifting-state-up.html) guide that covers this topic in more detail.